builtInFunctions/removeCodeLeaf.go

```diff
@@ -0,0 +1,85 @@
+package builtInFunctions
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const noOfArgsRemoveCodeLeaf = 1
+
+type removeCodeLeaf struct {
+	baseActiveHandler
+	accounts     vmcommon.AccountsAdapter
+	gasCost      uint64
+	mutExecution sync.RWMutex
+}
+
+// NewRemoveCodeLeafFunc creates a new removeCodeLeaf built-in function component
+func NewRemoveCodeLeafFunc(
+	gasCost uint64,
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+	accounts vmcommon.AccountsAdapter,
+) (*removeCodeLeaf, error) {
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+
+	mdt := &removeCodeLeaf{
+		gasCost:  gasCost,
+		accounts: accounts,
+	}
+
+	mdt.baseActiveHandler.activeHandler = func() bool {
+		return enableEpochsHandler.IsFlagEnabled(RemoveCodeLeafFlag)
+	}
+	return mdt, nil
+}
+
+// ProcessBuiltinFunction will remove trie code leaf corresponding to specified codeHash
+func (rcl *removeCodeLeaf) ProcessBuiltinFunction(
+	_, _ vmcommon.UserAccountHandler,
+	vmInput *vmcommon.ContractCallInput,
+) (*vmcommon.VMOutput, error) {
+	if vmInput == nil {
+		return nil, ErrNilVmInput
+	}
+	if len(vmInput.Arguments) != noOfArgsRemoveCodeLeaf {
+		return nil, fmt.Errorf("%w, expected %d, got %d ", ErrInvalidNumberOfArguments, noOfArgsRemoveCodeLeaf, len(vmInput.Arguments))
+	}
+
+	codeHash := vmInput.Arguments[0]
+
+	err := rcl.accounts.RemoveCodeLeaf(codeHash)
+	if err != nil {
+		return nil, err
+	}
+
+	gasRemaining := vmInput.GasProvided - rcl.gasCost
```

Comment:
This needs to be protected by mutex. RLock/RUnlock should do.

Reply:
done